sklearn/utils/tests/test_validation.py

```diff
@@ -324,6 +324,49 @@ def test_check_array_dtype_warning():
     assert_equal(X_checked.format, 'csr')
 
 
+def test_check_array_accept_sparse_type_exception():
+    X = [[1, 2], [3, 4]]
+    X_csr = sp.csr_matrix(X)
+    invalid_type = SVR()
+
+    msg = "A sparse matrix was passed, but dense data is required. " \
+        "Use X.toarray() to convert to a dense numpy array."
+    assert_raise_message(TypeError, msg,
+                         check_array, X_csr, accept_sparse=False)
+    assert_raise_message(TypeError, msg,
+                         check_array, X_csr, accept_sparse=None)
+
+    msg = "Parameter 'accept_sparse' should be a string, " \
+          "boolean or list of strings. You provided 'accept_sparse={}'."
+    assert_raise_message(ValueError, msg.format([]),
+                         check_array, X_csr, accept_sparse=[])
+    assert_raise_message(ValueError, msg.format(()),
+                         check_array, X_csr, accept_sparse=())
+    assert_raise_message(ValueError, msg.format(invalid_type),
+                         check_array, X_csr, accept_sparse=invalid_type)
+
+    msg = "'SVR' object"
+    assert_raise_message(TypeError, msg,
+                         check_array, X_csr, accept_sparse=[invalid_type])
+
+    # Test deprecation of 'None'
+    msg = "Passing None to parameter 'accept_sparse' is " \
+          "deprecated in 0.19. Use False instead."
+
+    assert_raise_message((DeprecationWarning, TypeError), msg,
```

Comment:
For warnings you'd use `assert_warns`, not `assert_raise_message`. And you're best to test with something that does not raise an exception (although `assert_warns(DeprecationError, assert_raise_message(TypeError, func, args))` should work also).